packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/SubscriptionImpl.kt

```diff
@@ -0,0 +1,89 @@
+package io.realm.mongodb.internal
+
+import io.realm.BaseRealm
+import io.realm.RealmInstant
+import io.realm.RealmObject
+import io.realm.TypedRealm
+import io.realm.dynamic.DynamicRealm
+import io.realm.dynamic.DynamicRealmObject
+import io.realm.internal.RealmInstantImpl
+import io.realm.internal.interop.RealmBaseSubscriptionSetPointer
+import io.realm.internal.interop.RealmInterop
+import io.realm.internal.interop.RealmSubscriptionPointer
+import io.realm.internal.platform.realmObjectCompanionOrThrow
+import io.realm.mongodb.sync.Subscription
+import io.realm.query.RealmQuery
+import kotlin.reflect.KClass
+
+internal class SubscriptionImpl(
+    private val realm: BaseRealm,
+    private val parentNativePointer: RealmBaseSubscriptionSetPointer,
+    internal val nativePointer: RealmSubscriptionPointer
+) : Subscription {
+    override val createdAt: RealmInstant
```

Comment:
Do these change over time? If not we could just assign them at construction time.